Control/Concurrent/Async.hs

```diff
@@ -432,6 +432,14 @@ pollSTM (Async _ w) = (Just <$> w) `orElse` return Nothing
 cancel :: Async a -> IO ()
 cancel a@(Async t _) = throwTo t AsyncCancelled <* waitCatch a
 
+-- | Cancel multiple asynchronous actions by throwing the @AsyncCancelled@
+-- exception to each of them in turn, then waiting for all the `Async` threads
+-- to complete.
+cancelMany :: [Async a] -> IO ()
+cancelMany as = do
+  mapM_ (\(Async t _) -> throwTo t AsyncCancelled) as
+  mapM_ waitCatch as
```

Comment:
👍 this will make a [a local definition I have for this](https://github.com/hspec/hspec/blob/44b0cbe4c0d16d43da908f3979ea4ff4c01e8986/hspec-core/src/Test/Hspec/Core/Runner/JobQueue.hs#L58-L64) redundant.